lsp/source/server.mts

```diff
@@ -155,15 +157,20 @@ connection.onInitialize(async (params: InitializeParams) => {
     };
   }
 
-  // TODO: currently only using the first workspace folder
-  const baseDir = params.workspaceFolders?.[0]?.uri.toString()
-  if (!baseDir)
-    throw new Error("Could not initialize without workspace folders")
+  if (!params.rootUri) {
```

Comment:
Although `rootUri` has been deprecated, it is not difficult for us to support it.

Reply:
Is this for compatibility with tools other than VSCode (e.g. tester or Vim) that aren't updated to the new interface? Otherwise, I don't see why we'd want to support a deprecated interface.

Reply:
In the future, there may be a demand for `Monaco` applied in browsers. For example, directly integrating civet-related language services in it. However, the version of `Monaco` will be relatively outdated. Maintaining compatibility will be very valuable here.

We can build some better online `playground`. Honestly, I think civet's work on types is very constructive. However, I didn't feel this in civet's `playground`. I think this is very regrettable.

Also, regarding unit tests, it seems that `workspaceFolders` are not needed. Here, I think the cost is not high and it can be simply supported.